packages/ui/components/form/select/components.tsx

```diff
@@ -52,7 +65,17 @@ export const OptionComponent = <
         props.isFocused && "dark:!bg-darkgray-200 !bg-gray-100",
         props.isSelected && "dark:!bg-darkgray-300 !bg-neutral-900"
       )}>
-      <span>{props.label}</span> {props.isSelected && <Icon.FiCheck className="h-4 w-4" />}
+      <>
+        <span className="mr-auto">{props.label}</span>
+        {(props.data as unknown as ExtendedOption).needsUpgrade && (
+          <Tooltip content={t("upgrade_to_enable_feature")}>
+            <button type="button" onClick={() => router.replace("/teams")}>
+              <Badge variant="gray">{t("upgrade")}</Badge>
+            </button>
```

Comment:
also should we turn this into a component so we can reuse this in other places?

Reply:
- Video Recordings @Udit-takkar 
- Disable Branding in settings

Reply:
Yes would probably make sense, I'll create a new component for it. Will open up a PR for it tomorrow